src/thunderbots/software/ai/navigator/util.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include "geom/util.h"
+#include "geom/point.h"
+
+/**
+  * returns the segment's final velocity (signed) after travelling from p1 to p2
+  * for a smooth transition to the p2 to p3 path, scaled by the ultimate final speed
+ */
```

Comment:
Could we change this to a full javadoc comment (short description, full description, `@param`'s, `@return`)

Reply:
Looks like we're still missing a function description at the start of the javadoc comment? Something like:
```
/**
 * Calculates the transition velocity for the robot between two line segments
 *
 * Calculates the velocity that the robot should be at when it is at the end of a 
 * given line segment in order to smoothly transition to another given line segment
 *
 * @param p1, p2, p3 are 3 points that define two line segments that form a path
 * @param final_vel is the intended final velocity at the end of the path
 * @return the first segment's final velocity (signed) after travelling from p1 to p2
 * for a smooth transition to the p2 to p3 path, scaled by the ultimate final speed
 */
```

Reply:
Also, is this returning velocity or speed? If it's speed, we should update the javadoc comment and function name accordingly

Reply:
it's velocity

Reply:
But it's a double? A velocity has direction (and so I'd expect a `Vector`), speed is a scalar (a `double`)?

Reply:
It should be speed. The value returned from this function will be used for the `final_speed` parameters when constructing Primitives. The purpose is to intelligently make those values non-zero (most of the time) so we don't stop in between each segment.

Reply:
should speed be only positive values then

Reply:
Yeah, that would make sense. You can't really enforce that with floating point numbers in C++, but we should def. make a note in the javadoc comment.

Reply:
ok